### PR TITLE
Add MockTransport for test environments

### PR DESCRIPTION
**Pull request by Slamdunk**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | -

This PR aims to provide mocking capability for test environments of the entire library

### CODE REVIEW

**Comment by Slamdunk** [comment]
Tests fail due to inaccessibility of live servers in Curl test cases.

**Comment by samdark** [comment]
Is the idea to use it in tests only?

**Comment by Slamdunk** [comment]
Exactly. We use this to mock real API of external providers, I thought this could be useful to others too.

**Comment by samdark** [comment]
Indeed, it looks useful. Would be great to get some documentation for it as well. Without docs it won't be noticed, I think.

**Comment by samdark** [comment]
@Slamdunk would you like to finish docs for it?

File: docs/guide/usage-transports.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ Predefined transports are:
  - [[\yii\httpclient\CurlTransport]] - sends HTTP messages using [Client URL Library (cURL)](http://php.net/manual/en/book.curl.php)
    This transport requires PHP 'curl' extension to be installed, but provides support for advanced features, like
    batch sending.
+ - [[\yii\httpclient\MockTransport]] - useful in test automation context, it doesn't send any real request and returns
+   the responses it is instructed to return.
```
Comment (Slamdunk) [inline docs/guide/usage-transports.md]: @samdark is this enough?

Comment (samdark) [reply docs/guide/usage-transports.md]: Yes.

Comment (samdark) [reply docs/guide/usage-transports.md]: Thank you!

**Comment by samdark** [comment]
All good. Would you please add a line for CHANGELOG.md?

**Comment by Slamdunk** [comment]
Done

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**